rustls/tests/api.rs

```diff
@@ -6367,6 +6367,48 @@ fn test_complete_io_with_no_io_needed() {
     );
 }
 
+#[test]
+fn test_junk_after_close_notify_received() {
```

Comment:
Do we need/want a separate test for unbuffered connections?

Reply:
Added one of these, the behaviour is different but better.